src/framegrab/cli/preview.py

```diff
@@ -0,0 +1,72 @@
+import shutil
+
+import ascii_magic
+import click
+import cv2
+import yaml
+from imgcat import imgcat
+from PIL import Image
+
+from framegrab import FrameGrabber
+
+
+def imgcat_preview(name: str, frame):
+    """Displays the given frame in the terminal using imgcat."""
+    imgcat(frame)
+
+
+def cv2_preview(name: str, frame):
+    """Displays the given frame in a cv2 window."""
+    cv2.imshow(name, frame)
+    key = cv2.waitKey(30)
+
+
+def ascii_preview(name: str, frame):
+    """Displays the given frame in the terminal using ascii art."""
+    columns, _ = shutil.get_terminal_size()
+    frame_rgb = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
+    pil_image = Image.fromarray(frame_rgb)
+    out = ascii_magic.from_pillow_image(pil_image)
+    out.to_terminal(columns=columns)
+
+
+def get_image_sources_from_config(config: str) -> list:
+    """Returns a dictionary of image sources from the given configuration file."""
+    with open(config, "r") as f:
+        configs = yaml.safe_load(f)
+    if "image_sources" not in configs:
+        raise click.BadParameter("Configuration file must contain an image_sources section.")
+    return configs["image_sources"]
+
+
+@click.command()
+# @click.argument('config', type=click.Path(exists=True), help="Path to the configuration file.")
+@click.argument("config", type=click.Path(exists=True))
+@click.argument(
+    "output",
+    type=click.Choice(["imgcat", "cv2", "ascii"], case_sensitive=False),
+    default="imgcat",
+)
+def preview(config: str, output: str):
+    """Previews images from each of the configured sources."""
+    image_sources = get_image_sources_from_config(config)
+    grabbers = FrameGrabber.create_grabbers(image_sources)
+
```

Comment:
should add time.sleep(1) here so that the camera (webcams in particular) have time to wake up. Otherwise the image will be very dark.

Reply:
I added this to the grabber itself, so the default behavior is what people wants.  It logs a note and is configurable if not wanted.